### PR TITLE
TINY-7895: Fix rounding errors in computed line height calculations

### PR DESCRIPTION
**Pull request by ashfordneil**
Related Ticket: TINY-7895

Description of Changes:
* Round computed font-size to 3 decimal places

Pre-checks:
* [x] Changelog entry added
* [x] ~Tests have been added (if applicable)~ they were already there, I just needed to enable them
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable): N/A

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/commands/LineHeight.ts [hunk 1]
```diff
@@ -17,7 +17,9 @@ export const lineHeightQuery = (editor: Editor): string => mapRange(editor, (elm
   const specifiedStyle = TransformFind.closest(elm, (elm) => Css.getRaw(elm, 'line-height'), Fun.curry(Compare.eq, root));
   const computedStyle = () => {
     // Css.get returns computed values (in px), and parseFloat will strip any non-number suffix
-    const lineHeight = parseFloat(Css.get(elm, 'line-height'));
+    const lineHeightRaw = parseFloat(Css.get(elm, 'line-height'));
+    // TINY-7895: Safari sometimes gives us 22.39999999999 and things like that
+    const lineHeight = parseFloat(lineHeightRaw.toPrecision(3));
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/commands/LineHeight.ts:22]: This isn't going to work in a number of cases, as take a slight deviation of the number you've given:
```
> (22.39899999999).toPrecision(3)
'22.4'
```
This instead probably needs to find up to 3 decimal places and round to that (e.g., not just rely on using only 3 values (I really do hate some JS apis, such as this). So in the above example, we probably want this to be `22.399` not `22.4`.

Note: I see I myself made this same flaw in `Transitions.ts` :facepalm:

**Review by lnewson — CHANGES_REQUESTED** [review]

**Comment by lnewson** [comment]
Also worth noting this has linter failures:
```
16:25:11  tinymce: /modules/tinymce/src/core/test/ts/browser/commands/LineHeightTest.ts
16:25:11  tinymce:   25:69  error  Prefer using arrow functions over plain functions  prefer-arrow/prefer-arrow-functions
16:25:11  tinymce: /modules/tinymce/src/themes/silver/test/ts/browser/editor/core/ChoiceControlsTest.ts
16:25:11  tinymce:   111:79  error  Prefer using arrow functions over plain functions  prefer-arrow/prefer-arrow-functions
16:25:11  tinymce: ✖ 2 problems (2 errors, 0 warnings)
```

File: modules/alloy/src/main/ts/ephox/alloy/positioning/view/Transitions.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ const shouldApplyTransitionCss = (transition: Transition, decision: RepositionDe
 
 const hasChanges = (position: PositionCss, intermediate: Record<TransitionProp, Optional<string>>): boolean => {
   // Round to 3 decimal points
-  const round = (value: string) => parseFloat(value).toPrecision(3);
+  const round = (value: string) => parseFloat(value).toFixed(3);
```
Comment (JamesToohey) [inline modules/alloy/src/main/ts/ephox/alloy/positioning/view/Transitions.ts:34]: Should we add a test to cover the `toPrecision` -> `toFixed` changes?

**Review by JamesToohey — APPROVED** [review]

File: modules/alloy/src/main/ts/ephox/alloy/positioning/view/Transitions.ts (hunk 2 above)
Comment (lnewson) [reply modules/alloy/src/main/ts/ephox/alloy/positioning/view/Transitions.ts:34]: I'm fine not including it atm, as testing that would be pretty involved/tricky and if it broke anything bad the existing tests would have caught it.

**Review by lnewson — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/commands/LineHeightTest.ts [hunk 3]
```diff
@@ -25,8 +25,7 @@ describe('browser.tinymce.core.commands.LineHeightTest', () => {
   });
 
   it('TINY-4843: Unspecified line-height can be read from element', function () {
-    // TODO: TINY-7895
-    if (platform.browser.isSafari()) {
+    if (platform.browser.isSafari() && (platform.browser.version.major < 15)) {
```
Comment (lnewson) [inline modules/tinymce/src/core/test/ts/browser/commands/LineHeightTest.ts:28]:
```suggestion
    if (platform.browser.isSafari() && platform.browser.version.major < 15) {
```

**Review by ltrouton — APPROVED** [review]

**Pull request merged**